### PR TITLE
feat(chat): chat customization telemetry

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Problem
- chat telemetry against codewhisperer service is missing customizationArn

## Solution
- add customizationArn in API call
- update model

## Functional Test
- verified request payload having expected customizationArn
```

VSC
// BuilderId
send chatAddMessage STE event with requestId=2e20dab9-461c-4cee-903d-fe24e7728440

send interactWithMessage STE event with requestId=237e4486-472a-4fd6-9e58-1ca0a58832ef

send chatUserModificationEvent STE event with requestId=2bae0f91-247e-4791-ae9e-9511bea6e441

//IdC with customization (NGDE-Custom_Fineturning-7K-V2
send chatAddMessage STE event with requestId=a2140652-1717-456b-b2cf-e6b76b8d8128

send interactWithMessage STE event with requestId=aba02e2d-8bc0-41d2-b1d0-991f0cd7a20d

send chatUserModificationEvent STE event with requestId=41df292d-a2b8-478d-a160-f49522419e68

// IdC without customization
send chatAddMessage STE event with requestId=ba38b07a-61df-4521-951c-c14f503f0781

send interactWithMessage STE event with requestId=78cfc617-20f0-4ebe-9046-11500b043a8d

send chatUserModificationEvent STE event with requestId=48880229-10fa-49e1-ba1a-7039067b9ec6

```

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/codewhisperer/client/codewhisperer.ts [hunk 1]
```diff
@@ -265,6 +265,8 @@ export class DefaultCodeWhispererClient {
         }
         const response = await (await this.createUserSdkClient()).sendTelemetryEvent(requestWithCommonFields).promise()
         getLogger().debug(`codewhisperer: sendTelemetryEvent requestID: ${response.$response.requestId}`)
+
+        return response
```
Comment (leigaol) [inline packages/core/src/codewhisperer/client/codewhisperer.ts:269]: is this change necessary?

Comment (Will-ShaoHua) [reply packages/core/src/codewhisperer/client/codewhisperer.ts:269]: i prefer having these so that callee can do whatever they want for the response like logging or something else

Comment (leigaol) [reply packages/core/src/codewhisperer/client/codewhisperer.ts:269]: OK I am fine as long as this does not cause regression

**Review by leigaol — APPROVED** [review]

**Review by andrewyuq — APPROVED** [review]

**Review by jpinkney-aws — APPROVED** [review]

File: packages/core/src/codewhispererChat/controllers/chat/telemetryHelper.ts [hunk 2]
```diff
@@ -258,6 +260,7 @@ export class CWCTelemetryHelper {
                         acceptedLineCount: event.cwsprChatAcceptedNumberOfLines,
                         acceptedSnippetHasReference: false,
                         hasProjectLevelContext: this.responseWithProjectContext.get(event.cwsprChatMessageId),
+                        customizationArn: undefinedIfEmpty(getSelectedCustomization().arn),
```
Comment (ege0zcan) [inline packages/core/src/codewhispererChat/controllers/chat/telemetryHelper.ts:263]: Does this correctly handle the case where:
1. user selects a customization X
2. user makes a chat request and gets a response back using customization X 
3. user selects a new customization Y
4. user interacts with message (e.g. copyToClipboard) on the last message that was received while customization X was active

If I understand the code correctly, we will record customization Y where we should actually record customizationX in this case

Comment (Will-ShaoHua) [reply packages/core/src/codewhispererChat/controllers/chat/telemetryHelper.ts:263]: you're right Ege, it won't handle the edge case you mentioned
